pkg/ext-proc/handlers/server.go

```diff
@@ -75,22 +77,30 @@ func (s *Server) Process(srv extProcPb.ExternalProcessor_ProcessServer) error {
 		var resp *extProcPb.ProcessingResponse
 		switch v := req.Request.(type) {
 		case *extProcPb.ProcessingRequest_RequestHeaders:
+			reqCtx.RequestReceivedTimestamp = time.Now()
 			resp = HandleRequestHeaders(reqCtx, req)
 			klog.V(3).Infof("Request context after HandleRequestHeaders: %+v", reqCtx)
 		case *extProcPb.ProcessingRequest_RequestBody:
 			resp, err = s.HandleRequestBody(reqCtx, req)
+			if err == nil {
+				metrics.RecordRequestCounter(reqCtx.Model, reqCtx.ResolvedTargetModel)
```

Comment:
Should we record the request counter regardless of the failure? The failure may result in an empty ResolvedTargetModel (which is fine I think, that just indicated the error occurred before we get the ResovledTargetModel)

Reply:
I added comments in the PR description.

The request counter only covers the success case now. A follow-up PR will be added to record request errors and update to record request counter for failure cases.